src/model.ts

```diff
@@ -802,6 +818,67 @@ export function build<TSchema extends BaseSchema, TOptions extends SchemaOptions
     throw new Error('findOneAndUpdate failed');
   });
 
+  /**
+   * @description
+   * Iterates over a result set, passing each found row into the iterator
+   * function.
+   *
+   * Useful when you want to process many rows without loading them all into
+   * memory at once.
+   *
+   * If the iterator function returns `false`, no more rows will be processed.
+   *
+   * @param filter {PaprFilter<TSchema>}
+   * @param [options] {FindOptions<TSchema>}
+   * @param iterator {function} - A callback of type `(row: TItem) => boolean | void`
+   *
+   * @example
+   * User.forEach(
+   *   { active: true, email: { $exists: true } },
+   *   { projection: { email: 1 } },
+   *   (row) => notify(row.email)
+   * )
+   */
+  model.forEach = wrap(
+    model,
+    async function forEach<
+      TProjection extends Projection<TSchema> | undefined,
+      TItem = TProjection extends undefined ? TSchema : ProjectionType<TSchema, TProjection>
+    >(
+      ...args:
```

Comment:
Yeah we were unable to do a typical overload signature because of how we're inlining/wrapping our methods. 

Just thinking out loud here - what if we just added a method to return the cursor from find? I'm afk so I can't immediately test it, but it looks like we'd be able to pass our projected schema as a generic to the Cursor. 

Ultimately your call, but I will say it keeps in line with the idea of Papr being a thin wrapper around Mongo.